jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketChannelState.java

```diff
@@ -98,7 +98,7 @@ public boolean isInputOpen()
     public boolean isOutputOpen()
     {
         State state = getState();
-        return (state==State.OPEN || state==State.ISHUT);
+        return (state==State.CONNECTED || state==State.OPEN || state==State.ISHUT);
```

Comment:
should we be allowed to send frames when we are connected but not open

Reply:
I believe so.  These states are inspired by the javax API and it does allows sending of frames in onOpen (aka CONNECTED state).